include/shell/shell_record.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
```

Comment:
there is already `shell_dummy` transport created for tests. Can you extend that one with record functionality?

Reply:
OK, done. I didn't know that was for testing. The name seems confusing.

Reply:
What name would be better in your opinion? I am not a native speaker hence the proposal. I can update the documentation to highlight this backend is for testing purposes.

Reply:
shell_fake I think, since it is a fake implementation